ur_robot_driver/src/hardware_interface.cpp

```diff
@@ -514,6 +513,12 @@ void URPositionHardwareInterface::asyncThread()
 hardware_interface::return_type URPositionHardwareInterface::read(const rclcpp::Time& time,
                                                                   const rclcpp::Duration& period)
 {
+  // We want to start the rtde comm the latest point possible due to the delay times arising from setting up the
+  // communication with multiple arms
+  if (!rtde_comm_has_been_started_) {
+    rtde_comm_has_been_started_ = true;
+    ur_driver_->startRTDECommunication();
```

Comment:
Logically, I believe it makes sense to swap those lines.
```suggestion
    ur_driver_->startRTDECommunication();
    rtde_comm_has_been_started_ = true;
```